### PR TITLE
Rework Hive image

### PR DESCRIPTION
**Pull request by lfrancke**
# Description

- Move postgresql and logging dependencies into build of Hive itself
- Backport all patches that have been backported on branch-3.1 upstream as well which affect the metastore
- Update patch dependencies
- Update Hadoop dependency to 3.3.6

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
- [ ] Add an entry to the CHANGELOG.md file
- [ ] Integration tests ran successfully
```

<details>
<summary>TIP: Running integration tests with a new product image</summary>

The image can be built and uploaded to the kind cluster with the following commands:

```shell
bake --product <product> --image-version <stackable-image-version>
kind load docker-image <image-tagged-with-the-major-version> --name=<name-of-your-test-cluster>
```

See the output of `bake` to retrieve the image tag for `<image-tagged-with-the-major-version>`.
</details>

### CODE REVIEW

**Comment by lfrancke** [comment]
I ran all integration tests and they all passed with the exception of the last one (random) which always failed at a Delete namespace step. Need to look into it.

**Comment by lfrancke** [comment]
We might want to wait with this until after 24.7
I'll let the release team decide.

**Comment by NickLarsenNZ** [comment]
Added as a stretch goal to https://github.com/stackabletech/issues/issues/557

File: hive/upload_new_postgres_driver_version.sh
Comment (sbernauer) [inline hive/upload_new_postgres_driver_version.sh:1]: Actually we want this for Nifi (https://github.com/stackabletech/decisions/issues/19), could you please move it to that folder as a preparation?
Otherwise no one will remember we ever had this script

File: hive/versions.py
Comment (sbernauer) [inline hive/versions.py:9]: The link is currently broken, https://github.com/apache/hadoop/blob/1be78238728da9266a4f88195058f08fd012bf9c/hadoop-project/pom.xml#L189 would be an alternative But in other versions.py we link to mvncentral, so I prefer to be consistent
```suggestion
        "aws_java_sdk_bundle": "1.12.367", # https://mvnrepository.com/artifact/org.apache.hadoop/hadoop-aws/3.3.6
```

File: hive/stackable/patches/3.1.3/series
Comment (sbernauer) [inline hive/stackable/patches/3.1.3/series:1]: I might be already there, but in case we are missing it:
Could you please document the command(s) to create this files from https://github.com/stackabletech/hive/commits/stackable/3.1.3/?
Maybe a README in the patches folder, also containing a link to https://github.com/stackabletech/hive?

File: hive/Dockerfile
Comment (sbernauer) [inline hive/Dockerfile:7]:
```suggestion
# Apache Hive up to 4.0.x(!) officially requires Java 8 (there is no distinction between building and running).
```

File: hive/Dockerfile
Comment (sbernauer) [inline hive/Dockerfile:65]: to be consistent with labels below
```suggestion
ARG NAME="Apache Hive metastore"
```

**Review by sbernauer — CHANGES_REQUESTED** [review]
I did not look at the content of the patches. In case I should review them as well pls ping me

File: hive/stackable/patches/3.1.3/series
Comment (lfrancke) [reply hive/stackable/patches/3.1.3/series:1]: @dervoeti and I tried. Unfortunately....there does not seem to be a way to do that using Stacked Git :(
It works locally only which is disappointing but we weren't able to find a better tool. I'll check if adding a README still makes sense.

File: hive/upload_new_postgres_driver_version.sh
Comment (lfrancke) [reply hive/upload_new_postgres_driver_version.sh:1]: For NiFi we want to do this differently as well I guess and include them via a pom.xml patch.

File: hive/versions.py
Comment (lfrancke) [reply hive/versions.py]: [`6221376` (#774)](https://github.com/stackabletech/docker-images/pull/774/commits/6221376e3306557ddc777639bebed75393ec8cfa)

File: hive/upload_new_postgres_driver_version.sh
Comment (lfrancke) [reply hive/upload_new_postgres_driver_version.sh:1]: I'd prefer to just throw it away.

Comment (sbernauer) [reply hive/upload_new_postgres_driver_version.sh:1]: Works for me

File: hive/stackable/patches/3.1.3/series
Comment (sbernauer) [reply hive/stackable/patches/3.1.3/series:1]: Honestly I didn't fully get it, but that's fine ;) But I think a readme pointing to https://github.com/stackabletech/hive/commits/stackable/3.1.3/ and https://github.com/stacked-git/stgit would be better than nothing :)

Comment (lfrancke) [reply hive/stackable/patches/3.1.3/series:1]: I did add a readme. Did I forget to commit it?

Comment (sbernauer) [reply hive/stackable/patches/3.1.3/series:1]: Sorry, I did not see that. Thanks!

**Review by sbernauer — APPROVED** [review]
I did not look at the content of the patches. In case I should review them as well pls ping me

**Comment by sbernauer** [comment]
Let's wait for https://github.com/stackabletech/docker-images/pull/803 first

**Review by sbernauer — APPROVED** [review]

**Pull request merged**
